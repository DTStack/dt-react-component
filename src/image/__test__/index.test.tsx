import React from 'react';
import { render, waitFor } from '@testing-library/react';

import Image from '..';
let originalImage: any = null;

describe('test Image', () => {
    beforeEach(() => {
        originalImage = window.Image;
        // @ts-ignore
        window.Image = class {
            constructor() {
                setTimeout(() => {
                    // @ts-ignore
                    this.onload();
                }, 100);
            }
        };
    });

    afterEach(() => {
        window.Image = originalImage;
    });

    it('should render correct', async () => {
        const wrapper = render(<Image src="https://example.com/image.jpg" />);
        await waitFor(() => {
            expect(wrapper).toMatchSnapshot();
        });
    });

    it('renders img when src is available', async () => {
        render(<Image src="https://example.com/image.jpg" />);
        await waitFor(() => {
            // 检查是否正确渲染 img 元素
            const imgDom = document.querySelector('img');
            expect(imgDom).not.toBeNull();
            expect(imgDom?.src).toEqual('https://example.com/image.jpg');
        });
    });

    it('renders img with correct props', async () => {
        const { container } = render(
            <Image
                src="https://example.com/image.jpg"
                height={200}
                width={100}
                className="test"
                style={{ backgroundColor: 'red' }}
            />
        );
        await waitFor(() => {
            // 检查是否正确渲染 img 元素
            const imgDom = container.querySelector('img');
            expect(imgDom).not.toBeNull();
            expect(imgDom?.height).toEqual(200);
            expect(imgDom?.width).toEqual(100);
            expect(imgDom?.className).toEqual('test');
            expect((imgDom?.style as any)._values).toEqual({ 'background-color': 'red' });
        });
    });
});
