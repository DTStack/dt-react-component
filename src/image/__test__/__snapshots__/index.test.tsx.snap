// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Image should render correct 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-busy="true"
        aria-live="polite"
        class="ant-spin ant-spin-spinning"
      >
        <span
          class="ant-spin-dot ant-spin-dot-spin"
        >
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
          <i
            class="ant-spin-dot-item"
          />
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-busy="true"
      aria-live="polite"
      class="ant-spin ant-spin-spinning"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
